backend/corpora/common/entities/dataset.py

```diff
@@ -1,9 +1,106 @@
 from .entity import Entity
-from ..corpora_orm import DbDataset
+from ..corpora_orm import DbDataset, DbDatasetArtifact, DbDeploymentDirectory, DbContributor, DbDatasetContributor
 
 
 class Dataset(Entity):
     table = DbDataset
 
     def __init__(self, db_object: DbDataset):
         super().__init__(db_object)
+
+    @classmethod
+    def create(
+        cls,
+        revision: int = 0,
+        name: str = "",
+        organism: str = "",
+        organism_ontology: str = "",
+        tissue: str = "",
+        tissue_ontology: str = "",
+        assay: str = "",
+        assay_ontology: str = "",
+        disease: str = "",
+        disease_ontology: str = "",
+        sex: str = "",
+        ethnicity: str = "",
+        ethnicity_ontology: str = "",
+        source_data_location: str = "",
+        preprint_doi: str = "",
+        publication_doi: str = "",
+        artifacts: list = None,
+        contributors: list = None,
+        deployment_directories: list = None,
+    ) -> "Dataset":
+        """
+        Creates a new dataset and related objects
+        :param revision:
+        :param name:
+        :param organism:
+        :param organism_ontology:
+        :param tissue:
+        :param tissue_ontology:
+        :param assay:
+        :param assay_ontology:
+        :param disease:
+        :param disease_ontology:
+        :param sex:
+        :param ethnicity:
+        :param ethnicity_ontology:
+        :param source_data_location:
+        :param preprint_doi:
+        :param publication_doi:
+        :param artifacts:
+        :param contributors:
+        :param deployment_directories:
+        :return:
+        """
+        uuid = cls.db.generate_id(DbDataset)
+
+        """
+        Prevent accidentally linking an existing row to a different Dataset. This maintains the relationship of one
+        to many for artifacts and deployment_directories
+        """
+        artifacts = artifacts if artifacts else []
```

Comment:
It's dangerous to set function defaults to mutable types such as a list. Run this a few times and see what happens to `a`:
```python
def foo(bar=[]):
    bar.append("stuff")
    return bar
```

This will lead to unexpected behavior.

Reply:
I think your approach is actually idiomatic, but you could also make the default `tuple()`

Reply:
@mckinsel oh nice tip

Reply:
I'm going to stick with how. While a tuple is immutable, it doesn't not have the same functionality as a list. That could cause problems further down the development line if we are assuming we are working with a list.